src/pages/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,280 @@
+- `mdx.smartypants` (`true` by default) to toggle SmartyPants in MDX. This replaces the `extendPlugins: 'defaults'` option.
+
+#### Added: `extendMarkdownConfig`
+
+Astro 2.0 adds the Boolean [`mdx.extendMarkdownConfig`](/en/guides/integrations-guide/mdx/#extendmarkdownconfig) option to configure whether or not MDX should inherit your Markdown configuration. This option is `true` by default, and MDX options will default to their equivalent in your Markdown config. You no longer need `extendPlugins: 'markdown'` in your configuration for this behavior.
+
+```diff
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  mdx: {
+-   extendPlugins: markdown,
+  }
+});
+```
+
+Replace `extendPlugins: defaults` with `extendMarkdownConfig: false` and use the individual options to enable GitHub-Flavored Markdown and SmartyPants if you want to inherit only Astro's default plugins, and not any other Markdown configuration options.
+
+```diff
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  mdx: {
+-   extendPlugins: defaults,
++   extendMarkdownConfig: false,
++   smartypants: true,
++   gfm: true,
+  }
+});
+```
+
+#### Added: More MDX config options to match Markdown
+
+Setting `extendMarkdownConfig` to false now allows you to individually set [every available Markdown configuration option](/en/reference/configuration-reference/#markdown-options) (except `drafts`) separately in your MDX integration config, including `syntaxHighlight`, `shikiConfig` and more.
+
+```astro
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
+
+export default defineConfig({
+  markdown: {
+    syntaxHighlight: 'prism',
+    remarkPlugins: [remarkPlugin1],
+    gfm: true,
+  },
+  integrations: [
+    mdx({
+      // `syntaxHighlight` inherited from Markdown
+
+      // Markdown `remarkPlugins` ignored,
+      // only `remarkPlugin2` applied.
+      remarkPlugins: [remarkPlugin2],
+      // `gfm` overridden to `false`
+      gfm: false,
+    })
+  ]
+});
+```
+
+### Added: Plugin access to frontmatter
+
+Astro 2.0 gives remark and rehype plugins access to user frontmatter via frontmatter injection. This means `data.astro.frontmatter` is now the _complete_ Markdown or MDX document's frontmatter, rather than an empty object.
+
+This allows plugin authors to modify existing frontmatter, or compute new properties based on other properties. For example, to compute a full image URL based on an `imageSrc` slug in your document frontmatter:
+
+  ```ts
+  export function remarkInjectSocialImagePlugin() {
+    return function (tree, file) {
+      const { frontmatter } = file.data.astro;
+      frontmatter.socialImageSrc = new URL(frontmatter.imageSrc, 'https://my-blog.com/').pathname;
+    };
+  }
+  ```
+
+### Added: `src/content/`
```

Comment:
nit: 

"Added: `src/content/` Reserved Folder"

or:

"Reserved: `src/content` for Content Collections"

As written, It sounds like this will be added to my project, or I need to add it myself?